gdrdrv/gdrdrv.c

```diff
@@ -314,6 +367,8 @@ static int gdrdrv_release(struct inode *inode, struct file *filp)
     }
     mutex_unlock(&info->lock);
 
+    filp->f_mapping = NULL;
```

Comment:
address_space_init_once seems to be an asymmetric API. No deinit is available. I guess that because address_space_init_once does not do memory allocation, the deinit API is not required.

Also,  setting filp->f_mapping back to the original mapping should be unnecessary. This is because filp is being destroyed after gdrdrv_release returns.

This method is inspired by nvidia-uvm. Please refer to the uvm_release function in uvm8.c.